requests-for-comments/2025-09-24-10x-engineering-onboarding.md

```diff
@@ -0,0 +1,88 @@
+3. **Internal poaching is breaking down** - we've moved people from overstaffed teams (Feature Flags) without communicating expectations upfront, creating confusion about hiring plans
+
+## Problem
+
+Current onboarding has too much variance in quality, operates in silos, doesn't teach the PostHog way effectively, creates an "onboarding tax" on high-performing teams, and makes inefficient use of our technical debt.
+
+**More details:**
+
+1. **Variance in quality.** Some teams have well-thought-out processes, others wing it. New hires get wildly different experiences depending on which team they join, making it impossible to assess performance consistently or ensure everyone learns our core ways of working.
+
+2. **Teams operate in silos.** New engineers only meet their immediate team members during their first weeks. This limits cross-team collaboration and means people miss out on learning from other parts of the product and codebase.
+
+3. **Not teaching the PostHog way effectively.** Our core values around autonomy, ownership, and development practices are best learned through experience, but we're trying to teach them within the narrow context of a single product area.
+
+4. **"Onboarding tax" on high-performing teams.** Teams with strong onboarding culture (e.g. Feature Flags) end up overhiring, expecting that ramped-up engineers will get poached by teams that need them. This creates weird expectations—how do teams know they're supposed to overhire? It also unfairly burdens teams that invested in good onboarding processes.
+
+5. **Inefficient use of starter tasks.** We have technical debt and starter tasks across all products, but currently these only get tackled by people on those specific teams. We're missing an opportunity to pay down this debt while giving new hires diverse learning experiences.
+
+6. **Scheduling headache for offsites.** Coordinating onboarding offsites is a logistical pain when people start at different times and are spread across teams.
+
+## Proposal: HogBoarding
+
+**Core principle:** Teach the PostHog way first, then specialize.
+
+**The system:** New engineers onboard in cohorts of 3-6 people for a standardized 2-3 week period, then transition to their specific teams with a solid foundation.
+
+**Implementation details:**
+
+- **Cohort formation:** Everyone starting within a 2-week window onboards together
+- **Rotating leadership:** If 4 product teams are hiring, any mixture of those team leads can run onboarding (spreads load, prevents burnout, ensures knowledge sharing)
+- **Standardized curriculum:** Development practices, architecture overview, core tools
+- **Cross-product work:** Rotating pool of "good first issues" across all products
+- **Pair programming and code review sessions** with onboarding buddies who work together rather than in isolation
+- **Cross-team social activities and team introductions**
+- **Cohort-based offsites** that are easier to schedule and more cost-effective
```

Comment:
I think having consistent onboarding plans, including locations, hotels, weworks/offices, etc would be a good thing to have